internal/pkg/sourceinfer/analyzer.go

```diff
@@ -224,9 +231,13 @@ func inferSources(pass *analysis.Pass, conf *config.Config, objGraph objectGraph
 		if seen[o] {
 			continue
 		}
-		if !(isSourceType(conf, o.Type()) || pass.ImportObjectFact(o, &inferredSourceFact{})) {
+
+		if v, ok := o.(*types.Var); !(ok && taggedFields.IsSource(v)) &&
+			!isSourceType(conf, o.Type()) &&
+			!pass.ImportObjectFact(o, &inferredSourceFact{}) {
```

Comment:
This conditional has gotten pretty heavy.  Consider using De Morgan's to collapse the negations, i.e.

```golang
		if v, ok := o.(*types.Var); !(
			ok && taggedFields.IsSource(v) ||
				isSourceType(conf, o.Type()) ||
				pass.ImportObjectFact(o, &inferredSourceFact{})) {
			continue
		}
```

Reply:
Interestingly, when I wrote the conditional it started out in the form that you're suggesting, and then I used De Morgan's to turn it into what it is now. I think it's more readable as it is, but it's not very readable in either case.

I think this PR #173 will help alleviate this.